consensus/fast-sync/src/fast_sync.rs

```diff
@@ -149,6 +207,87 @@ async fn validate_hashes(
     })
 }
 
+async fn validate_block<C>(
+    mut context_svc: C,
+    block: Box<Block>,
+    txs: HashMap<[u8; 32], Transaction>,
+    token: ValidBlockId,
+) -> Result<FastSyncResponse, FastSyncError>
+where
+    C: Service<
+            BlockChainContextRequest,
+            Response = BlockChainContextResponse,
+            Error = tower::BoxError,
+        > + Send
+        + 'static,
+    C::Future: Send + 'static,
+{
+    let BlockChainContextResponse::Context(checked_context) = context_svc
+        .ready()
+        .await?
+        .call(BlockChainContextRequest::GetContext)
+        .await?
+    else {
+        panic!("Context service returned wrong response!");
+    };
+
+    let block_chain_ctx = checked_context.unchecked_blockchain_context().clone();
+
+    let block_hash = block.hash();
+    if block_hash != token.0 {
+        return Err(FastSyncError::BlockHashMismatch);
+    }
+
+    let block_blob = block.serialize();
+
+    let Some(Input::Gen(height)) = block.miner_tx.prefix.inputs.first() else {
+        return Err(FastSyncError::MinerTx(MinerTxError::InputNotOfTypeGen));
+    };
+
+    let txs_vec: Vec<Transaction> = txs.values().cloned().collect();
+    let mut txs = Vec::<VerifiedTransactionInformation>::with_capacity(txs_vec.len());
+    for tx in txs_vec {
+        let data = TransactionVerificationData::new(tx)?;
+        txs.push(VerifiedTransactionInformation {
+            tx_blob: data.tx_blob,
+            tx_weight: data.tx_weight,
+            fee: data.fee,
+            tx_hash: data.tx_hash,
+            tx: data.tx,
+        });
+    }
+
+    let total_fees = txs.iter().map(|tx| tx.fee).sum::<u64>();
+
+    let weight = block.miner_tx.weight() + txs.iter().map(|tx| tx.tx_weight).sum::<usize>();
+
+    let generated_coins = check_miner_tx(
+        &block.miner_tx,
+        total_fees,
+        block_chain_ctx.chain_height,
+        weight,
+        block_chain_ctx.median_weight_for_block_reward,
+        block_chain_ctx.already_generated_coins,
+        &block_chain_ctx.current_hf,
+    )?;
```

Comment:
We could sum the outputs here instead of calling `check_miner_tx`:
```suggestion
    let total_fees = txs.iter().map(|tx| tx.fee).sum::<u64>();
    let total_outputs = block
        .miner_tx
        .prefix
        .outputs
        .iter()
        .map(|output| output.amount.unwrap_or(0))
        .sum::<u64>();

    let generated_coins = total_outputs - total_fees;

    let weight = block.miner_tx.weight() + txs.iter().map(|tx| tx.tx_weight).sum::<usize>();
```